File: src/ast/LocalName.java
package ast;

import java.io.StringWriter;

public class LocalName extends BaseNode {
    private BaseNode _encoding;
    private BaseNode _entity;

    public LocalName(BaseNode encoding, BaseNode entity) {
        super(NodeType.LocalName);
        _encoding = encoding;
        _entity = entity;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        _encoding.Print(writer);
        writer.write("::");
        _entity.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/LocalName.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class LocalName(private val encoding: BaseNode, private val entity: BaseNode) : BaseNode(NodeType.LocalName) {
    public override fun printLeft(writer: StringWriter) {
        encoding.print(writer)
        writer.write("::")
        entity.print(writer)
    }

}